2020-11-14 00:56:06.182 | DEBUG    | aria2p.api:remove:450 - Try to remove download 0000000000000001
2020-11-14 00:56:06.188 | SUCCESS  | aria2p.api:remove:457 - Removed download 0000000000000001
2020-11-14 00:56:06.192 | DEBUG    | aria2p.api:remove:462 - Failed to remove download result 0000000000000001
2020-11-14 00:56:06.192 | TRACE    | aria2p.api:remove:463 - GID 0000000000000001 is not found
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw0-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7f1502471cf8>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7f1502428730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
    └ <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
    │    └ <function WorkerPool._perform_spawn at 0x7f15024288c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7f1502428620>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7f15023c5ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7f150100c2e8>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7f150100c2e8>]
          │         └ <function pytest_cmdline_main at 0x7f15010ec7b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7f15010ec950>
           │            └ <_pytest.config.Config object at 0x7f150100c2e8>
           └ <function wrap_session at 0x7f15010ec840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │       │            │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    │       │            └ <function _main at 0x7f15010ec950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
          └ <HookImpl plugin_name='139728100656744', plugin=<__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [972]
    │    └ <function WorkerInteractor.run_one_test at 0x7f1501002268>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_pause_method[4]>
    │    │      │    │                            └ <Function test_remove_files_not_complete[0]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>, <Function test_pause_method[4]>]
          │         └ <function pytest_runtest_protocol at 0x7f1500b4a1e0>
          └ <HookImpl plugin_name='rerunfailures', plugin=<module 'pytest_rerunfailures' from '/home/pawamoy/.cache/pypoetry/virtualenvs/...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
    reports = runtestprotocol(item, nextitem=nextitem, log=False)
              │               │              └ <Function test_pause_method[4]>
              │               └ <Function test_remove_files_not_complete[0]>
              └ <function runtestprotocol at 0x7f15010e22f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ False
    │       │      │               └ <Function test_remove_files_not_complete[0]>
    │       │      └ <function call_and_report at 0x7f15010e2620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_api.py::test_remove_files_not_complete[0]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_remove_files_not_complete[0]>
           └ <function call_runtest_hook at 0x7f15010e2840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7f14f90cc400>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_runtest_call at 0x7f15010e2488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7f15010a80d0>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_remove_files_not_complete[0]>
    │    └ <property object at 0x7f150113a7c8>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_pyfunc_call at 0x7f150109b2f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'), 'port': 15255}
             └ <function test_remove_files_not_complete at 0x7f14fb7bac80>

  File "/media/data/dev/pawamoy/aria2p/tests/test_api.py", line 176, in test_remove_files_not_complete
    assert server.api.remove(downloads, files=True)
           │      │   │      └ [<aria2p.downloads.Download object at 0x7f14f8e54898>, <aria2p.downloads.Download object at 0x7f14f8e54470>]
           │      │   └ <function API.remove at 0x7f14fbccb048>
           │      └ API(Client(host='http://localhost', port=15255, secret='********'))
           └ <tests.conftest._Aria2Server object at 0x7f14f8e2ad68>

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 460, in remove
    self.client.remove_download_result(download.gid)
    │    │      │                      │        └ <property object at 0x7f14fbcb9408>
    │    │      │                      └ <aria2p.downloads.Download object at 0x7f14f8e54898>
    │    │      └ <function Client.remove_download_result at 0x7f14fbd3b158>
    │    └ Client(host='http://localhost', port=15255, secret='********')
    └ API(Client(host='http://localhost', port=15255, secret='********'))

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 1459, in remove_download_result
    return self.call(self.REMOVE_DOWNLOAD_RESULT, [gid])
           │    │    │    │                        └ '0000000000000001'
           │    │    │    └ 'aria2.removeDownloadResult'
           │    │    └ Client(host='http://localhost', port=15255, secret='********')
           │    └ <function Client.call at 0x7f14fbdb2c80>
           └ Client(host='http://localhost', port=15255, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 261, in call
    return self.res_or_raise(self.post(self.get_payload(method, params, msg_id=msg_id)))
           │    │            │    │    │    │           │       │              └ None
           │    │            │    │    │    │           │       └ ['0000000000000001']
           │    │            │    │    │    │           └ 'aria2.removeDownloadResult'
           │    │            │    │    │    └ <staticmethod object at 0x7f14fbd8e5f8>
           │    │            │    │    └ Client(host='http://localhost', port=15255, secret='********')
           │    │            │    └ <function Client.post at 0x7f14fbdb2ea0>
           │    │            └ Client(host='http://localhost', port=15255, secret='********')
           │    └ <staticmethod object at 0x7f14fbd8e668>
           └ Client(host='http://localhost', port=15255, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 391, in res_or_raise
    raise Client.response_as_exception(response)
          │      │                     └ {'id': -1, 'jsonrpc': '2.0', 'error': {'code': 1, 'message': 'GID 0000000000000001 is not found'}}
          │      └ <staticmethod object at 0x7f14fbdc86d8>
          └ <class 'aria2p.client.Client'>

aria2p.client.ClientException: GID 0000000000000001 is not found
2020-11-14 00:56:06.208 | DEBUG    | aria2p.api:remove:479 - aria2 control file /tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0/5m.aria2 was not found
2020-11-14 00:56:06.208 | WARNING  | aria2p.downloads:root_files_paths:345 - Can't determine file path '.' relative to '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.208 | TRACE    | aria2p.downloads:root_files_paths:346 - '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw0-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7f1502471cf8>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7f1502428730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
    └ <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
    │    └ <function WorkerPool._perform_spawn at 0x7f15024288c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7f1502428620>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7f15023c5ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7f150100c2e8>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7f150100c2e8>]
          │         └ <function pytest_cmdline_main at 0x7f15010ec7b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7f15010ec950>
           │            └ <_pytest.config.Config object at 0x7f150100c2e8>
           └ <function wrap_session at 0x7f15010ec840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │       │            │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    │       │            └ <function _main at 0x7f15010ec950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
          └ <HookImpl plugin_name='139728100656744', plugin=<__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [972]
    │    └ <function WorkerInteractor.run_one_test at 0x7f1501002268>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_pause_method[4]>
    │    │      │    │                            └ <Function test_remove_files_not_complete[0]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>, <Function test_pause_method[4]>]
          │         └ <function pytest_runtest_protocol at 0x7f1500b4a1e0>
          └ <HookImpl plugin_name='rerunfailures', plugin=<module 'pytest_rerunfailures' from '/home/pawamoy/.cache/pypoetry/virtualenvs/...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
    reports = runtestprotocol(item, nextitem=nextitem, log=False)
              │               │              └ <Function test_pause_method[4]>
              │               └ <Function test_remove_files_not_complete[0]>
              └ <function runtestprotocol at 0x7f15010e22f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ False
    │       │      │               └ <Function test_remove_files_not_complete[0]>
    │       │      └ <function call_and_report at 0x7f15010e2620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_api.py::test_remove_files_not_complete[0]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_remove_files_not_complete[0]>
           └ <function call_runtest_hook at 0x7f15010e2840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7f14f90cc400>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_runtest_call at 0x7f15010e2488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7f15010a80d0>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_remove_files_not_complete[0]>
    │    └ <property object at 0x7f150113a7c8>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_pyfunc_call at 0x7f150109b2f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'), 'port': 15255}
             └ <function test_remove_files_not_complete at 0x7f14fb7bac80>

  File "/media/data/dev/pawamoy/aria2p/tests/test_api.py", line 176, in test_remove_files_not_complete
    assert server.api.remove(downloads, files=True)
           │      │   │      └ [<aria2p.downloads.Download object at 0x7f14f8e54898>, <aria2p.downloads.Download object at 0x7f14f8e54470>]
           │      │   └ <function API.remove at 0x7f14fbccb048>
           │      └ API(Client(host='http://localhost', port=15255, secret='********'))
           └ <tests.conftest._Aria2Server object at 0x7f14f8e2ad68>

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 484, in remove
    self.remove_files([download], force=True)
    │    │             └ <aria2p.downloads.Download object at 0x7f14f8e54898>
    │    └ <staticmethod object at 0x7f14fbd06710>
    └ API(Client(host='http://localhost', port=15255, secret='********'))

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 695, in remove_files
    for path in download.root_files_paths:
                │        └ <property object at 0x7f14fbcb93b8>
                └ <aria2p.downloads.Download object at 0x7f14f8e54898>

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/downloads.py", line 343, in root_files_paths
    relative_path = file.path.relative_to(self.dir)
                    │    │                │    └ <property object at 0x7f14fbcb9f48>
                    │    │                └ <aria2p.downloads.Download object at 0x7f14f8e54898>
                    │    └ <property object at 0x7f14fbcb9098>
                    └ <aria2p.downloads.File object at 0x7f14f8e54320>

  File "/home/pawamoy/.basher-packages/pyenv/pyenv/versions/3.6.9/lib/python3.6/pathlib.py", line 874, in relative_to
    .format(str(self), str(formatted)))
                │          └ '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
                └ PosixPath('.')

ValueError: '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.220 | DEBUG    | aria2p.api:remove:450 - Try to remove download 0000000000000002
2020-11-14 00:56:06.222 | SUCCESS  | aria2p.api:remove:457 - Removed download 0000000000000002
2020-11-14 00:56:06.225 | DEBUG    | aria2p.api:remove:462 - Failed to remove download result 0000000000000002
2020-11-14 00:56:06.225 | TRACE    | aria2p.api:remove:463 - GID 0000000000000002 is not found
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw0-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7f1502471cf8>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7f1502428730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
    └ <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
    │    └ <function WorkerPool._perform_spawn at 0x7f15024288c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7f1502428620>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7f15023c5ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7f150100c2e8>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7f150100c2e8>]
          │         └ <function pytest_cmdline_main at 0x7f15010ec7b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7f15010ec950>
           │            └ <_pytest.config.Config object at 0x7f150100c2e8>
           └ <function wrap_session at 0x7f15010ec840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │       │            │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    │       │            └ <function _main at 0x7f15010ec950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
          └ <HookImpl plugin_name='139728100656744', plugin=<__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [972]
    │    └ <function WorkerInteractor.run_one_test at 0x7f1501002268>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_pause_method[4]>
    │    │      │    │                            └ <Function test_remove_files_not_complete[0]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>, <Function test_pause_method[4]>]
          │         └ <function pytest_runtest_protocol at 0x7f1500b4a1e0>
          └ <HookImpl plugin_name='rerunfailures', plugin=<module 'pytest_rerunfailures' from '/home/pawamoy/.cache/pypoetry/virtualenvs/...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
    reports = runtestprotocol(item, nextitem=nextitem, log=False)
              │               │              └ <Function test_pause_method[4]>
              │               └ <Function test_remove_files_not_complete[0]>
              └ <function runtestprotocol at 0x7f15010e22f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ False
    │       │      │               └ <Function test_remove_files_not_complete[0]>
    │       │      └ <function call_and_report at 0x7f15010e2620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_api.py::test_remove_files_not_complete[0]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_remove_files_not_complete[0]>
           └ <function call_runtest_hook at 0x7f15010e2840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7f14f90cc400>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_runtest_call at 0x7f15010e2488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7f15010a80d0>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_remove_files_not_complete[0]>
    │    └ <property object at 0x7f150113a7c8>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_pyfunc_call at 0x7f150109b2f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'), 'port': 15255}
             └ <function test_remove_files_not_complete at 0x7f14fb7bac80>

  File "/media/data/dev/pawamoy/aria2p/tests/test_api.py", line 176, in test_remove_files_not_complete
    assert server.api.remove(downloads, files=True)
           │      │   │      └ [<aria2p.downloads.Download object at 0x7f14f8e54898>, <aria2p.downloads.Download object at 0x7f14f8e54470>]
           │      │   └ <function API.remove at 0x7f14fbccb048>
           │      └ API(Client(host='http://localhost', port=15255, secret='********'))
           └ <tests.conftest._Aria2Server object at 0x7f14f8e2ad68>

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 460, in remove
    self.client.remove_download_result(download.gid)
    │    │      │                      │        └ <property object at 0x7f14fbcb9408>
    │    │      │                      └ <aria2p.downloads.Download object at 0x7f14f8e54470>
    │    │      └ <function Client.remove_download_result at 0x7f14fbd3b158>
    │    └ Client(host='http://localhost', port=15255, secret='********')
    └ API(Client(host='http://localhost', port=15255, secret='********'))

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 1459, in remove_download_result
    return self.call(self.REMOVE_DOWNLOAD_RESULT, [gid])
           │    │    │    │                        └ '0000000000000002'
           │    │    │    └ 'aria2.removeDownloadResult'
           │    │    └ Client(host='http://localhost', port=15255, secret='********')
           │    └ <function Client.call at 0x7f14fbdb2c80>
           └ Client(host='http://localhost', port=15255, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 261, in call
    return self.res_or_raise(self.post(self.get_payload(method, params, msg_id=msg_id)))
           │    │            │    │    │    │           │       │              └ None
           │    │            │    │    │    │           │       └ ['0000000000000002']
           │    │            │    │    │    │           └ 'aria2.removeDownloadResult'
           │    │            │    │    │    └ <staticmethod object at 0x7f14fbd8e5f8>
           │    │            │    │    └ Client(host='http://localhost', port=15255, secret='********')
           │    │            │    └ <function Client.post at 0x7f14fbdb2ea0>
           │    │            └ Client(host='http://localhost', port=15255, secret='********')
           │    └ <staticmethod object at 0x7f14fbd8e668>
           └ Client(host='http://localhost', port=15255, secret='********')

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/client.py", line 391, in res_or_raise
    raise Client.response_as_exception(response)
          │      │                     └ {'id': -1, 'jsonrpc': '2.0', 'error': {'code': 1, 'message': 'GID 0000000000000002 is not found'}}
          │      └ <staticmethod object at 0x7f14fbdc86d8>
          └ <class 'aria2p.client.Client'>

aria2p.client.ClientException: GID 0000000000000002 is not found
2020-11-14 00:56:06.236 | DEBUG    | aria2p.api:remove:479 - aria2 control file /tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0/6m.aria2 was not found
2020-11-14 00:56:06.236 | WARNING  | aria2p.downloads:root_files_paths:345 - Can't determine file path '.' relative to '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.236 | TRACE    | aria2p.downloads:root_files_paths:346 - '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw0-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7f1502471cf8>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7f1502428730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
    └ <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
    │    └ <function WorkerPool._perform_spawn at 0x7f15024288c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7f1502428620>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7f15023c5ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7f150100c2e8>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7f150100c2e8>]
          │         └ <function pytest_cmdline_main at 0x7f15010ec7b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7f15010ec950>
           │            └ <_pytest.config.Config object at 0x7f150100c2e8>
           └ <function wrap_session at 0x7f15010ec840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │       │            │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    │       │            └ <function _main at 0x7f15010ec950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
          └ <HookImpl plugin_name='139728100656744', plugin=<__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [972]
    │    └ <function WorkerInteractor.run_one_test at 0x7f1501002268>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_pause_method[4]>
    │    │      │    │                            └ <Function test_remove_files_not_complete[0]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>, <Function test_pause_method[4]>]
          │         └ <function pytest_runtest_protocol at 0x7f1500b4a1e0>
          └ <HookImpl plugin_name='rerunfailures', plugin=<module 'pytest_rerunfailures' from '/home/pawamoy/.cache/pypoetry/virtualenvs/...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
    reports = runtestprotocol(item, nextitem=nextitem, log=False)
              │               │              └ <Function test_pause_method[4]>
              │               └ <Function test_remove_files_not_complete[0]>
              └ <function runtestprotocol at 0x7f15010e22f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ False
    │       │      │               └ <Function test_remove_files_not_complete[0]>
    │       │      └ <function call_and_report at 0x7f15010e2620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_api.py::test_remove_files_not_complete[0]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_remove_files_not_complete[0]>
           └ <function call_runtest_hook at 0x7f15010e2840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7f14f90cc400>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_runtest_call at 0x7f15010e2488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7f15010a80d0>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_remove_files_not_complete[0]>
    │    └ <property object at 0x7f150113a7c8>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_pyfunc_call at 0x7f150109b2f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'), 'port': 15255}
             └ <function test_remove_files_not_complete at 0x7f14fb7bac80>

  File "/media/data/dev/pawamoy/aria2p/tests/test_api.py", line 176, in test_remove_files_not_complete
    assert server.api.remove(downloads, files=True)
           │      │   │      └ [<aria2p.downloads.Download object at 0x7f14f8e54898>, <aria2p.downloads.Download object at 0x7f14f8e54470>]
           │      │   └ <function API.remove at 0x7f14fbccb048>
           │      └ API(Client(host='http://localhost', port=15255, secret='********'))
           └ <tests.conftest._Aria2Server object at 0x7f14f8e2ad68>

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 484, in remove
    self.remove_files([download], force=True)
    │    │             └ <aria2p.downloads.Download object at 0x7f14f8e54470>
    │    └ <staticmethod object at 0x7f14fbd06710>
    └ API(Client(host='http://localhost', port=15255, secret='********'))

  File "/media/data/dev/pawamoy/aria2p/src/aria2p/api.py", line 695, in remove_files
    for path in download.root_files_paths:
                │        └ <property object at 0x7f14fbcb93b8>
                └ <aria2p.downloads.Download object at 0x7f14f8e54470>

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/downloads.py", line 343, in root_files_paths
    relative_path = file.path.relative_to(self.dir)
                    │    │                │    └ <property object at 0x7f14fbcb9f48>
                    │    │                └ <aria2p.downloads.Download object at 0x7f14f8e54470>
                    │    └ <property object at 0x7f14fbcb9098>
                    └ <aria2p.downloads.File object at 0x7f14f8e542b0>

  File "/home/pawamoy/.basher-packages/pyenv/pyenv/versions/3.6.9/lib/python3.6/pathlib.py", line 874, in relative_to
    .format(str(self), str(formatted)))
                │          └ '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
                └ PosixPath('.')

ValueError: '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.247 | WARNING  | aria2p.downloads:root_files_paths:345 - Can't determine file path '.' relative to '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.247 | TRACE    | aria2p.downloads:root_files_paths:346 - '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw0-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7f1502471cf8>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7f1502428730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
    └ <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
    │    └ <function WorkerPool._perform_spawn at 0x7f15024288c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7f1502428620>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7f15023c5ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7f150100c2e8>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7f150100c2e8>]
          │         └ <function pytest_cmdline_main at 0x7f15010ec7b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7f15010ec950>
           │            └ <_pytest.config.Config object at 0x7f150100c2e8>
           └ <function wrap_session at 0x7f15010ec840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │       │            │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    │       │            └ <function _main at 0x7f15010ec950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
          └ <HookImpl plugin_name='139728100656744', plugin=<__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [972]
    │    └ <function WorkerInteractor.run_one_test at 0x7f1501002268>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_pause_method[4]>
    │    │      │    │                            └ <Function test_remove_files_not_complete[0]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>, <Function test_pause_method[4]>]
          │         └ <function pytest_runtest_protocol at 0x7f1500b4a1e0>
          └ <HookImpl plugin_name='rerunfailures', plugin=<module 'pytest_rerunfailures' from '/home/pawamoy/.cache/pypoetry/virtualenvs/...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
    reports = runtestprotocol(item, nextitem=nextitem, log=False)
              │               │              └ <Function test_pause_method[4]>
              │               └ <Function test_remove_files_not_complete[0]>
              └ <function runtestprotocol at 0x7f15010e22f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ False
    │       │      │               └ <Function test_remove_files_not_complete[0]>
    │       │      └ <function call_and_report at 0x7f15010e2620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_api.py::test_remove_files_not_complete[0]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_remove_files_not_complete[0]>
           └ <function call_runtest_hook at 0x7f15010e2840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7f14f90cc400>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_runtest_call at 0x7f15010e2488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7f15010a80d0>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_remove_files_not_complete[0]>
    │    └ <property object at 0x7f150113a7c8>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_pyfunc_call at 0x7f150109b2f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'), 'port': 15255}
             └ <function test_remove_files_not_complete at 0x7f14fb7bac80>

  File "/media/data/dev/pawamoy/aria2p/tests/test_api.py", line 178, in test_remove_files_not_complete
    for file in download.root_files_paths:
                │        └ <property object at 0x7f14fbcb93b8>
                └ <aria2p.downloads.Download object at 0x7f14f8e54898>

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/downloads.py", line 343, in root_files_paths
    relative_path = file.path.relative_to(self.dir)
                    │    │                │    └ <property object at 0x7f14fbcb9f48>
                    │    │                └ <aria2p.downloads.Download object at 0x7f14f8e54898>
                    │    └ <property object at 0x7f14fbcb9098>
                    └ <aria2p.downloads.File object at 0x7f14f8e54320>

  File "/home/pawamoy/.basher-packages/pyenv/pyenv/versions/3.6.9/lib/python3.6/pathlib.py", line 874, in relative_to
    .format(str(self), str(formatted)))
                │          └ '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
                └ PosixPath('.')

ValueError: '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.257 | WARNING  | aria2p.downloads:root_files_paths:345 - Can't determine file path '.' relative to '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
2020-11-14 00:56:06.257 | TRACE    | aria2p.downloads:root_files_paths:346 - '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "<string>", line 8, in <module>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1554, in serve
    SlaveGateway(io=io, id=id, _startcount=2).serve()
    │               │      └ 'gw0-slave'
    │               └ <execnet.gateway_base.Popen2IO object at 0x7f1502471cf8>
    └ <class 'execnet.gateway_base.SlaveGateway'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1060, in serve
    self._execpool.integrate_as_primary_thread()
    │    │         └ <function WorkerPool.integrate_as_primary_thread at 0x7f1502428730>
    │    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
    └ <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 267, in integrate_as_primary_thread
    self._perform_spawn(reply)
    │    │              └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
    │    └ <function WorkerPool._perform_spawn at 0x7f15024288c8>
    └ <execnet.gateway_base.WorkerPool object at 0x7f150241a940>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 285, in _perform_spawn
    reply.run()
    │     └ <function Reply.run at 0x7f1502428620>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 220, in run
    self._result = func(*args, **kwargs)
    │              │     │       └ {}
    │              │     └ ((<Channel id=1 open>, ('"""\n    This module is executed in remote subprocesses and helps to\n    control a remote testing s...
    │              └ <bound method SlaveGateway.executetask of <execnet.gateway_base.SlaveGateway object at 0x7f150241a9e8>>
    └ <execnet.gateway_base.Reply object at 0x7f15023c3828>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/execnet/gateway_base.py", line 1084, in executetask
    do_exec(co, loc)  # noqa
    │       │   └ {'channel': <Channel id=1 open>, '__name__': '__channelexec__', '__builtins__': {'__name__': 'builtins', '__doc__': "Built-in...
    │       └ <code object <module> at 0x7f15023c5ed0, file "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/...
    └ <built-in function exec>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 237, in <module>
    config.hook.pytest_cmdline_main(config=config)
    │      │    │                          └ <_pytest.config.Config object at 0x7f150100c2e8>
    │      │    └ <_HookCaller 'pytest_cmdline_main'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_cmdline_main'>
           │    │         └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'config': <_pytest.config.Config object at 0x7f150100c2e8>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_cmdline_main'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                │    │    │    │                     └ <_HookCaller 'pytest_cmdline_main'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1500ffccf8>
                └ <_HookCaller 'pytest_cmdline_main'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<_pytest.config.Config object at 0x7f150100c2e8>]
          │         └ <function pytest_cmdline_main at 0x7f15010ec7b8>
          └ <HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-p...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 306, in pytest_cmdline_main
    return wrap_session(config, _main)
           │            │       └ <function _main at 0x7f15010ec950>
           │            └ <_pytest.config.Config object at 0x7f150100c2e8>
           └ <function wrap_session at 0x7f15010ec840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 257, in wrap_session
    session.exitstatus = doit(config, session) or 0
    │       │            │    │       └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │       │            │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    │       │            └ <function _main at 0x7f15010ec950>
    │       └ <ExitCode.OK: 0>
    └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/main.py", line 313, in _main
    config.hook.pytest_runtestloop(session=session)
    │      │    │                          └ <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>
    │      │    └ <_HookCaller 'pytest_runtestloop'>
    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    └ <_pytest.config.Config object at 0x7f150100c2e8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtestloop'>
           │    │         └ <_HookCaller 'pytest_runtestloop'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'session': <Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>}
           │    │               │     └ [<HookImpl plugin_name='main', plugin=<module '_pytest.main' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-...
           │    │               └ <_HookCaller 'pytest_runtestloop'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                │    │    │    │                     └ <_HookCaller 'pytest_runtestloop'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007f28>
                └ <_HookCaller 'pytest_runtestloop'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Session tests exitstatus=<ExitCode.OK: 0> testsfailed=2 testscollected=4690>]
          │         └ <bound method WorkerInteractor.pytest_runtestloop of <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
          └ <HookImpl plugin_name='139728100656744', plugin=<__channelexec__.WorkerInteractor object at 0x7f14fbce0668>>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 70, in pytest_runtestloop
    self.run_one_test(torun)
    │    │            └ [972]
    │    └ <function WorkerInteractor.run_one_test at 0x7f1501002268>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/xdist/remote.py", line 87, in run_one_test
    self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
    │    │      │    │                            │              └ <Function test_pause_method[4]>
    │    │      │    │                            └ <Function test_remove_files_not_complete[0]>
    │    │      │    └ <_HookCaller 'pytest_runtest_protocol'>
    │    │      └ <pluggy.hooks._HookRelay object at 0x7f15010c7fd0>
    │    └ <_pytest.config.Config object at 0x7f150100c2e8>
    └ <__channelexec__.WorkerInteractor object at 0x7f14fbce0668>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_protocol'>
           │    │         └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>, 'nextitem': <Function test_pause_method[4]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_protocol'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_protocol'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007dd8>
                └ <_HookCaller 'pytest_runtest_protocol'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>, <Function test_pause_method[4]>]
          │         └ <function pytest_runtest_protocol at 0x7f1500b4a1e0>
          └ <HookImpl plugin_name='rerunfailures', plugin=<module 'pytest_rerunfailures' from '/home/pawamoy/.cache/pypoetry/virtualenvs/...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
    reports = runtestprotocol(item, nextitem=nextitem, log=False)
              │               │              └ <Function test_pause_method[4]>
              │               └ <Function test_remove_files_not_complete[0]>
              └ <function runtestprotocol at 0x7f15010e22f0>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 127, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
    │       │      │               │             └ False
    │       │      │               └ <Function test_remove_files_not_complete[0]>
    │       │      └ <function call_and_report at 0x7f15010e2620>
    │       └ <method 'append' of 'list' objects>
    └ [<TestReport 'test_api.py::test_remove_files_not_complete[0]' when='setup' outcome='passed'>]
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 216, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
           │                 │     │       └ {}
           │                 │     └ 'call'
           │                 └ <Function test_remove_files_not_complete[0]>
           └ <function call_runtest_hook at 0x7f15010e2840>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in call_runtest_hook
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       │           │             └ (<class '_pytest.outcomes.Exit'>, <class 'KeyboardInterrupt'>)
            │          │       │           └ 'call'
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 310, in from_call
    result = func()  # type: Optional[TResult]
             └ <function call_runtest_hook.<locals>.<lambda> at 0x7f14f90cc400>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 256, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
            │          │       └ {}
            │          └ <Function test_remove_files_not_complete[0]>
            └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_runtest_call'>
           │    │         └ <_HookCaller 'pytest_runtest_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'item': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_runtest_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                │    │    │    │                     └ <_HookCaller 'pytest_runtest_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': False, 'historic': False}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007ba8>
                └ <_HookCaller 'pytest_runtest_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_runtest_call at 0x7f15010e2488>
          └ <HookImpl plugin_name='runner', plugin=<module '_pytest.runner' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/runner.py", line 163, in pytest_runtest_call
    item.runtest()
    │    └ <function Function.runtest at 0x7f15010a80d0>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 1627, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
    │    │                                   └ <Function test_remove_files_not_complete[0]>
    │    └ <property object at 0x7f150113a7c8>
    └ <Function test_remove_files_not_complete[0]>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
           │    │         │     │    │                └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │         │     │    └ <function _HookCaller.get_hookimpls at 0x7f15012fc158>
           │    │         │     └ <_HookCaller 'pytest_pyfunc_call'>
           │    │         └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <bound method PluginManager._hookexec of <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>>
           └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
           │    │               │     │        └ {'pyfuncitem': <Function test_remove_files_not_complete[0]>}
           │    │               │     └ [<HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgx...
           │    │               └ <_HookCaller 'pytest_pyfunc_call'>
           │    └ <function PluginManager.__init__.<locals>.<lambda> at 0x7f15010027b8>
           └ <_pytest.config.PytestPluginManager object at 0x7f1501afa6d8>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
                │    │    │    │                     │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                │    │    │    │                     └ <_HookCaller 'pytest_pyfunc_call'>
                │    │    │    └ <method 'get' of 'dict' objects>
                │    │    └ {'firstresult': True, 'historic': False, 'warn_on_impl': None}
                │    └ <pluggy.hooks.HookSpec object at 0x7f1501007908>
                └ <_HookCaller 'pytest_pyfunc_call'>
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
          │         │         └ [<Function test_remove_files_not_complete[0]>]
          │         └ <function pytest_pyfunc_call at 0x7f150109b2f0>
          └ <HookImpl plugin_name='python', plugin=<module '_pytest.python' from '/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQ...
  File "/home/pawamoy/.cache/pypoetry/virtualenvs/aria2p-h4kgxQkF-py3.6/lib/python3.6/site-packages/_pytest/python.py", line 184, in pytest_pyfunc_call
    result = testfunction(**testargs)
             │              └ {'tmp_path': PosixPath('/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'), 'port': 15255}
             └ <function test_remove_files_not_complete at 0x7f14fb7bac80>

  File "/media/data/dev/pawamoy/aria2p/tests/test_api.py", line 178, in test_remove_files_not_complete
    for file in download.root_files_paths:
                │        └ <property object at 0x7f14fbcb93b8>
                └ <aria2p.downloads.Download object at 0x7f14f8e54470>

> File "/media/data/dev/pawamoy/aria2p/src/aria2p/downloads.py", line 343, in root_files_paths
    relative_path = file.path.relative_to(self.dir)
                    │    │                │    └ <property object at 0x7f14fbcb9f48>
                    │    │                └ <aria2p.downloads.Download object at 0x7f14f8e54470>
                    │    └ <property object at 0x7f14fbcb9098>
                    └ <aria2p.downloads.File object at 0x7f14f8e542b0>

  File "/home/pawamoy/.basher-packages/pyenv/pyenv/versions/3.6.9/lib/python3.6/pathlib.py", line 874, in relative_to
    .format(str(self), str(formatted)))
                │          └ '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
                └ PosixPath('.')

ValueError: '.' does not start with '/tmp/pytest-of-pawamoy/pytest-67/popen-gw0/test_remove_files_not_complete0'
